File: app/src/playstore/java/deltazero/amarok/utils/AppCenterUtil.java
package deltazero.amarok.utils;

import android.app.Application;

import com.microsoft.appcenter.AppCenter;
import com.microsoft.appcenter.crashes.Crashes;

import deltazero.amarok.PrefMgr;
import kotlin.NotImplementedError;

public class AppCenterUtil {

    private static final String appSecret = "6bcd9547-9df2-4023-bfcd-6e1a0f0f9e12";

    public static void cleanUpdatePostpone() {
        throw new NotImplementedError("Not available in Play Store version.");
    }

    public static void checkUpdate() {
        throw new NotImplementedError("Not available in Play Store version.");
    }

    public static void setAnalyticsEnabled(boolean enabled) {
        Crashes.setEnabled(enabled);
    }

    public static boolean isAnalyticsEnabled() {
        return Crashes.isEnabled().get();
    }

    public static void startAppCenter(Application application) {
        assert PrefMgr.initialized;
        AppCenter.start(application, appSecret, Crashes.class);
    }

    public static boolean isAvailable() {
        return false;
    }
}

File: app/src/playstore/java/deltazero/amarok/utils/AppCenterUtil.kt
package deltazero.amarok.utils

import android.app.Application
import com.microsoft.appcenter.AppCenter
import com.microsoft.appcenter.crashes.Crashes

object AppCenterUtil {
  private const val appSecret = "6bcd9547-9df2-4023-bfcd-6e1a0f0f9e12"

  @JvmStatic
  fun cleanUpdatePostpone() {
    throw NotImplementedError("Not available in Play Store version.")
  }

  @JvmStatic
  fun checkUpdate() {
    throw NotImplementedError("Not available in Play Store version.")
  }

  @JvmStatic
  fun setAnalyticsEnabled(enabled: Boolean) {
    Crashes.setEnabled(enabled)
  }

  @JvmStatic fun isAnalyticsEnabled(): Boolean = Crashes.isEnabled().get()

  @JvmStatic
  fun startAppCenter(
    application: Application,
    autoUpdateEnabled: Boolean,
    onDisableAutoUpdate: Runnable,
  ) {
    AppCenter.start(application, appSecret, Crashes::class.java)
  }

  @JvmStatic fun isAvailable(): Boolean = false
}
